Templates/colors.json

```diff
@@ -0,0 +1,4687 @@
+        { "name": "Connect", "type": "bool" }
+      ]
+    },
+    "ObjLayerBoardSpawner": {
+      "fields": [
+        { "name": "CollisionWidth", "type": "float32" },
+        { "name": "CollisionHeight", "type": "float32" },
+        { "name": "OnLayer0", "type": "uint8" },
+        { "name": "OnLayer1", "type": "uint8" },
+        { "name": "OffLayer0", "type": "uint8" },
+        { "name": "OffLayer1", "type": "uint8" },
+        { "name": "IsBackEffective", "type": "bool" }
+      ]
+    },
+    "ObjPointMarkerSpawner": {
+      "fields": [
+        { "name": "Width", "type": "float32" },
+        { "name": "Height", "type": "float32" },
+        { "name": "StageType", "type": "ObjPointMarkerSpawner::StageType" }
+      ]
+    },
+    "ObjShadowSpawner": {
+      "fields": [
+        { "name": "Width", "type": "float32" },
+        { "name": "Height", "type": "float32" },
+        { "name": "Depth", "type": "float32" }
+      ]
+    },
+    "TestObjSpawner": {
+      "fields": [
+      ]
+    },
+    "CCTTestObjSpawner": {
+      "fields": [
+      ]
+    },
+    "ObjAppearThornBallSpawner": {
+      "fields": [
+        { "name": "Time", "type": "float32" },
+        { "name": "Distance", "type": "float32" }
+      ]
+    },
+    "ObjChainSpikeBallSpawner": {
+      "fields": [
+        { "name": "Radius", "type": "float32" },
+        { "name": "AngularSpeed", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "RotateType", "type": "ObjChainSpikeBallSpawner::RotateType" }
+      ]
+    },
+    "ObjChaserBombSpawner": {
+      "fields": [
+        { "name": "ThresholdStubmle", "type": "float32" },
+        { "name": "IsBigChaserMode", "type": "bool" }
+      ]
+    },
+    "ObjDirectionalThornSpawner": {
+      "fields": [
+        { "name": "OffTime", "type": "float32" },
+        { "name": "OnTime", "type": "float32" },
+        { "name": "MoveTime", "type": "float32" },
+        { "name": "ForetasteTime", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "TargetID", "type": "object_reference" },
+        { "name": "UsePanel", "type": "bool" },
+        { "name": "Type", "type": "ObjDirectionalThornSpawner::Type" }
+      ]
+    },
+    "ObjFlameSpawner": {
+      "fields": [
+        { "name": "OffTime", "type": "float32" },
+        { "name": "OnTime", "type": "float32" },
+        { "name": "LengthType", "type": "ObjFlameSpawner::LengthType" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "Type", "type": "ObjFlameSpawner::Type" }
+      ]
+    },
+    "ObjGrindThornSpawner": {
+      "fields": [
+      ]
+    },
+    "ObjMovingThornBallSpawner": {
+      "fields": [
+        { "name": "Time", "type": "float32" },
+        { "name": "Distance", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "Direction", "type": "ObjMovingThornBallSpawner::Direction" }
+      ]
+    },
```

Comment:
Parameters named based on `ObjThornBallMoveParam` in Sonic Forces. Set data parser function (0x8008CF74, NTSC-J Wii) used for additional help.